ui/src/app/cluster-workflow-templates/components/cluster-workflow-template-details/cluster-workflow-template-details.tsx

```diff
@@ -38,20 +40,15 @@ export class ClusterWorkflowTemplateDetails extends BasePage<RouteComponentProps
     }
 
     public componentDidMount(): void {
-        services.info
-            .getInfo()
-            .then(info => this.setState({namespace: info.managedNamespace || 'default'}))
-            .catch(error => this.setState(error));
         services.clusterWorkflowTemplate
             .get(this.name)
             .then(template => this.setState({template}))
+            .then(() => services.info.getInfo())
+            .then(info => this.setState({namespace: info.managedNamespace || Utils.getCurrentNamespace() || 'default'}))
```

Comment:
cluster templates an the one type where we cannot infer the namespace from the URL